### PR TITLE
feat(gha): init ci

### LINKED ISSUE #13: ci/cd is missing, so tests are not run
TO quote @line-o "we have tests, we ran them as well"
now they run all the time against `:latest` 

What node versions should be included for CI?
What exist versions?

To automate deployment to GH packages someone needs to add valid GH credentials for releases (probably define the `.xar` as release artifact so it gets attached to the release automatically.

!!WARNING!!
semanticRelease plugin works by adhering to commit message style 

(we could even configure a commit check on github actions to ensure that only valid commit messages are being used by all contributors)

### PR DESCRIPTION
**Pull request by duncdrum**
close #13 

@line-o your turn

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          fetch-depth: 0
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Install dependencies
+        run: npm ci
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (duncdrum) [inline .github/workflows/node.js.yml:46]: #14

**Review by line-o — COMMENTED** [review]
Thank you very much to contribute this!
I do not think we need to test the router with different versions of node js, though :)

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
```
Comment (duncdrum) [inline .github/workflows/node.js.yml:38]: @line-o this should match the node version for which we run tests

Comment (line-o) [reply .github/workflows/node.js.yml:38]: ok, let's do this 👍

**Comment by line-o** [comment]
How would we approach testing on different existdb versions?
Can we define the matrix strategy earlier and have services per step?

File: expath-pkg.xml.tmpl [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <package xmlns="http://expath.org/ns/pkg" name="@namespace@" abbrev="@target@" version="@version@" spec="1.0">
+    <!-- <dependency processor="http://exist-db.org" semver-min="5.0.0" /> -->
```
Comment (duncdrum) [inline expath-pkg.xml.tmpl:3]: see #15 this line should not be commented out, (did i overlook a typo)?

**Comment by duncdrum** [comment]
ci testing found the first bug (#15)
exist matrix expansion works
i slightly edited the readme to include more info, and to pull the basic installation information up front, where it belongs imv before the how to use this app section
I would think that a node matrix expansion makes sense to test against upcoming node versions, but that's up to the maintainers

**Comment by line-o** [comment]
@duncdrum Oh OK, so then it is settled, roaster is exist 5+ only. Then we can remove that from the matrix.
And 💯 for adding that, too :)

**Review by line-o — APPROVED** [review]
Very good work

**Comment by line-o** [comment]
@duncdrum have you seen files or folders being part of the package that should not be there?

**Comment by line-o** [comment]
@duncdrum the XAR package of roaster is the contents of the `build` folder. I just checked, everything fine, nothing to filter there.

**Comment by duncdrum** [comment]
@line-o no I didn’t check the contents just saw a copy instruction starting from `.` which made me wonder. 

It also just occurred to me that once semantic release is configured in this repo, we should put a note into the readme so that ppl can lookup the commit message conventions

**Comment by line-o** [comment]
@duncdrum Yes, we should mention that in a "Contribute" section in the readme, perhaps.

Do you have an idea how to add the `xar` artefact to the release?

**Comment by duncdrum** [comment]
I’m looking into it…

**Comment by line-o** [comment]
@duncdrum I am not against it. Though, there was no need to do that in the past, since that can be enforced in the PR by humans as well 
;)

**Comment by duncdrum** [comment]
ok this about covers it from my end. Feel free to rebase or squash. I can't really say if everything is working without releases being generated, which i don't want to do from my fork, it needs to work here. 

there are no checks for commit message style from CI, since you prefer the human touch :-)

**Review by line-o — APPROVED** [review]
Brilliant

**Comment by line-o** [comment]
> I didn't check if the gulp build command cleans up after itself? otherwise running the release plugin locally might attach multiple .xars to a new release if old ones are lying around.

Well, then that could happen :) - the XARs are left there on purpose.

**Review by line-o — APPROVED** [review]
"Development" and "Testing" are now subsections of "Contributing". I think that's OK.

**Pull request merged**